lib/archethic/p2p/message/start_mining.ex

```diff
@@ -4,75 +4,120 @@ defmodule Archethic.P2P.Message.StartMining do
+    case check_synchronization(network_chains_view_hash, p2p_view_hash) do
+      :ok ->
+        with {:election, true} <- {:election, Mining.valid_election?(tx, validation_nodes)},
+             {:elected, true} <-
+               {:elected, Enum.any?(validation_nodes, &(&1 == Crypto.last_node_public_key()))},
+             {:mining, false} <- {:mining, Mining.processing?(tx.address)} do
+          {:ok, _} = Mining.start(tx, welcome_node_public_key, validation_nodes)
+          %Ok{}
+        else
+          {:election, false} ->
+            Logger.error("Invalid validation node election",
+              transaction_address: Base.encode16(tx.address),
+              transaction_type: tx.type
+            )
 
-        %Error{reason: :network_issue}
+            %Error{reason: :network_issue}
 
-      {:elected, false} ->
-        Logger.error("Unexpected start mining message",
-          transaction_address: Base.encode16(tx.address),
-          transaction_type: tx.type
-        )
+          {:elected, false} ->
+            Logger.error("Unexpected start mining message",
+              transaction_address: Base.encode16(tx.address),
+              transaction_type: tx.type
+            )
+
+            %Error{reason: :network_issue}
 
-        %Error{reason: :network_issue}
+          {:mining, true} ->
+            Logger.warning("Transaction already in mining process",
+              transaction_address: Base.encode16(tx.address),
+              transaction_type: tx.type
+            )
 
-      {:mining, true} ->
-        Logger.warning("Transaction already in mining process",
+            %Ok{}
+        end
+
+      {:error, sync_issue} ->
+        Logger.warning("Current node may be out of synchronization: #{inspect(sync_issue)}",
           transaction_address: Base.encode16(tx.address),
           transaction_type: tx.type
         )
 
-        %Ok{}
+        case sync_issue do
+          :network_chains_sync ->
+            SelfRepair.resync_all_network_chains()
+
+          :p2p_sync ->
+            SelfRepair.resync_p2p()
+
+          :both_sync ->
+            SelfRepair.resync_all_network_chains()
+            SelfRepair.resync_p2p()
+        end
+
+        %Error{reason: sync_issue}
     end
   end
 
   @spec serialize(t()) :: bitstring()
   def serialize(%__MODULE__{
         transaction: tx,
         welcome_node_public_key: welcome_node_public_key,
-        validation_node_public_keys: validation_node_public_keys
+        validation_node_public_keys: validation_node_public_keys,
+        network_chains_view_hash: network_chains_view_hash,
+        p2p_view_hash: p2p_view_hash
       }) do
     <<Transaction.serialize(tx)::binary, welcome_node_public_key::binary,
       length(validation_node_public_keys)::8,
-      :erlang.list_to_binary(validation_node_public_keys)::binary>>
+      :erlang.list_to_binary(validation_node_public_keys)::binary,
+      byte_size(network_chains_view_hash)::8, network_chains_view_hash::binary,
```

Comment:
If you use `:crypto.hash` as previously mentionned, you can remove the byte size as the hash will always be 256 bits